concepts.md

```diff
@@ -91,6 +91,150 @@ An implementation may formally validate the structure of a CWL document using
 SALAD schemas located at
 https://github.com/common-workflow-language/common-workflow-language/tree/master/v1.1.0-dev1
 
+### map
+
+Note: This section is non-normative.
+> type: array&lt;ComplexType&gt; |  
+> map&lt;`key_field`, ComplexType&gt;
+
+The above syntax in the CWL specifications means there are two or more ways to write the given value.
+
+Option one is a array and is the most verbose option.
+In our example here we use the generic`ComplexType`, but
+in reality it would be one of `InputRecordField`, `OutputRecordField`,
```

Comment:
Technically the mapping attaches to the field, not the types of the field, I'm not exactly sure if listing these out here is helpful enough to outweigh the burden of adding another place in the text that has to be maintained every time we update the spec.